Add wording permitting TTML3 and a modular approach.

nigelmegitt opened the pull request:
Closes #35.

nigelmegitt committed:
Add wording permitting TTML3 and a modular approach.

Closes #35.

nigelmegitt committed:
Merge branch 'gh-pages' into issue-35-ttml-modularisation

skynavga commented on index.html:
```diff
@@ -125,10 +125,17 @@ <h2>Scope</h2>
+          <li>Publish a new <a href="https://www.w3.org/TR/ttml3/">Timed
+              Text Markup Language 3 (TTML3)</a> specification, consisting of 
+              a series of modules where each module and any base document
+              is a Recommendation. In the process
```
Change

>consisting of
>a series of modules where each module and any base document
>is a Recommendation.

to read

>a Recommendation that serves as a base document, and which
>refers to one or more functional modules, each to be published
>as a separate Recommendation.

nigelmegitt replied:
Done.

skynavga commented on index.html:
```diff
@@ -125,10 +125,17 @@ <h2>Scope</h2>
+              is a Recommendation. In the process
+              of producing this new revision, the Group SHOULD:
             <ol type="a">
+              <li>Include a framework defining how each module integrates with
```
Change "each module integrates" to "modules integrate".

nigelmegitt replied:
Done.

skynavga commented on index.html:
```diff
@@ -188,7 +195,12 @@ <h2>Scope</h2>
+          <p>In order to advance to 
+            <a href="https://www.w3.org/Consortium/Process/#RecsPR"
+            title="Proposed Recommendation">Proposed Recommendation</a>,
+            each specification is expected to have
```
Change "specification" to "specification, i.e., either base document or module document,".

nigelmegitt replied:
I'm going to tackle this a slightly different way, to refer to Recommendation track technical reports instead, which cover all the specification documents we publish without getting into the weeds of whether each is a specification, a module, a base, or whatever.

skynavga commented on index.html:
```diff
@@ -208,6 +220,29 @@ <h2> Deliverables </h2>
+            <dd>
+              <p class="draft-status"><b>Draft state:</b> <i><a href="https://w3c.github.io/ttml3/">Editor's
+                    Draft</a></i></p>
+              <p class="milestone"><b>Expected completion: [Q4-2019]</b> </p>
```
Change "Q4-2019" to "Q3-2020". The goal is still the former, but we should build in some wiggle room.

nigelmegitt replied:
This is only setting an expectation, so there's adequate wriggle room already. I would rather not weaken that expectation right now. Happy to discuss on Thursday's call.

skynavga commented on index.html:
```diff
@@ -223,7 +258,8 @@ <h3> Normative Specifications </h3>
               Text Markup Language 2 (TTML2), <a href="https://www.w3.org/TR/2018/CR-ttml2-20180313/">Candidate Recommendation, 13 March
-              2018.</a>
+              2018</a>.
+
```
We should update this to refer to a new 2nd edition WD (or ED).

nigelmegitt replied:
The deliverables do need updating, but not in this pull request, which is tackling modularisation for TTML3 only. I'll tackle this elsewhere - see https://github.com/w3c/charter-timed-text/pull/42#pullrequestreview-215521606

nigelmegitt committed:
Address pull request comments by @skynavga

nigelmegitt committed:
editorial tidying of success criteria

Avoid duplicated text by using a list.

nigelmegitt commented on index.html:
```diff
@@ -208,6 +224,29 @@ <h2> Deliverables </h2>
+              text media for the purpose of interchange among authoring,
+              distribution and playback systems. Timed text is textual
+              information that is intrinsically or extrinsically associated with
+              timing information. </dd>
```
This may consist of one or more documents. Can be not-necessarily TTML3, leave to WG to decide.

nigelmegitt replied:
Done.

nigelmegitt commented on index.html:
```diff
@@ -125,10 +125,18 @@ <h2>Scope</h2>
-          <li>Publish a Recommendation for a new <a href="https://www.w3.org/TR/ttml2/">Timed
-              Text Markup Language 2 (TTML2)</a> specification. In the process
-            of producing this new revision, the Group SHOULD:
+          <li>Publish a new <a href="https://www.w3.org/TR/ttml3/">Timed
```
Just new versions of TTML as needed.

nigelmegitt replied:
Done.

nigelmegitt commented on index.html:
```diff
@@ -208,6 +224,29 @@ <h2> Deliverables </h2>
+                    Draft</a></i></p>
+              <p class="milestone"><b>Expected completion: [Q4-2019]</b> </p>
+            </dd>
+            <dd><b>Adopted Working Draft:</b> Timed
```
W3 team has tools for generating adopted WDs.

nigelmegitt reviewed (commented):
Also need to add any Module registry to the list of potential Notes.

nigelmegitt committed:
Address comments raised during TTWG meeting 2019-03-21:

* Be non-specific about TTML version number;
* Remove specific details about modules and frameworks (without 
prohibiting them);
* Add a MAY list to permit specification of new functionality or 
movement of existing functionality in new Recommendations;
* Refer to TTML3 as a collection of specifications.
* Add "Registries" to the list of Other Deliverables.

skynavga commented on index.html:
```diff
@@ -208,6 +225,29 @@ <h2> Deliverables </h2>
+            <dt class="spec"><a href="https://www.w3.org/TR/ttml3/">Timed Text
+                Markup Language 3 (TTML3)<br>
+              </a></dt>
+            <dd>This collection of specifications defines a content type that
```
Change to "specification". All of the content of this item refers to an Editor's Draft (in the singular) which is not a collection. TTML3 is a single REC track document (=specification) deliverable. If you want, you can a new item after  Timed Text Markup Language 3 (TTML3) that has a title "Timed Text Markup Language 3 (TTML3) Modules", and then list the ones we expect to publish along with the TTML3 specification.

nigelmegitt replied:
Done.

skynavga commented on index.html:
```diff
@@ -208,6 +225,29 @@ <h2> Deliverables </h2>
+            <dd>
+              <p class="draft-status"><b>Draft state:</b> <i><a href="https://w3c.github.io/ttml3/">Editor's
+                    Draft</a></i></p>
+              <p class="milestone"><b>Expected completion: [Q4-2019]</b> </p>
```
I'm not comfortable saying Q4 is "expected". While that may be our goal, I would be more comfortable saying "Q3-2020" for expected.

nigelmegitt replied:
@skynavga to do that would be a divergence from the working mode we agreed to adopt, in which features that we could not get through the process during this year would be deferred. If we do not expect to publish until Q3 2020 then I question whether we should even start. There was a lot of pressure caused by how long TTML2 took, and I think there was agreement to do things differently next time.

skynavga replied:
@nigelmegitt now that you have updated the deliverable info for TTML3 to one spec, it is possible to meet that schedule; however, I have doubts that all the issues/requirements that we have delegated to modules will complete

nigelmegitt committed:
Attempt to address @skynavga's review comments with a slightly different approach:
* Change TTML3 to be a singular specification;
* Add a new paragraph to the Deliverables section setting out the WG's 
expectation to publish new versions of TTML annually and to publish TTML 
features in new Recommendations as per the scope.

nigelmegitt commented:
@skynavga I've attempted to address your review comments - if we are going to change the expected publication date of TTML3 we need a WG discussion about that because I think the group has clear expectations about that already.
